Stratis.Bitcoin/Wallet/Wallet.cs

```diff
@@ -387,7 +442,20 @@ public HdAddress()
         /// </returns>
         public bool IsChangeAddress()
         {
-            return int.Parse(this.HdPath.Split('/')[4]) == 1;
+            if (string.IsNullOrWhiteSpace(this.HdPath))
+                return false;
```

Comment:
maybe throw an exception here.
we don't support non-HD addresses so there should always be an address here.
if not, something is wrong and we need to know about it.